### PR TITLE
feat!: Updated AWS provider to v4, added ECR repo force_delete argument in docker-build module

### PR DESCRIPTION
**Pull request by 1Mill**
## Description

Maintain behavior to force delete created `aws_ecr_repository` even if it contains images. 

## Motivation and Context

[AWS v4.22.0](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#4220-july--8-2022) introduced a breaking change to the behavior of this module. 

Previously, `aws_ecr_repository` resources would always be destroyed, even if they contained images. Now, `aws_ecr_repository` resources are only destroyed if either (a) the repository is completely empty of images or (b) `force_delete` is set to `true`. 

This PR sets `force_delete` to `true` so that any created `aws_ecr_repository` resource is always successfully destroyed on `terraform destroy` to maintain the existing behavior of this module. 

## Breaking Changes

* Requires `aws >= 4.22.0`

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

### CODE REVIEW

File: modules/docker-build/main.tf [hunk 1]
```diff
@@ -24,6 +24,7 @@ resource "docker_registry_image" "this" {
 resource "aws_ecr_repository" "this" {
   count = var.create_ecr_repo ? 1 : 0
 
+  force_delete         = true
```
Comment (antonbabenko) [inline modules/docker-build/main.tf:27]: Please add `force_delete` variable with default value `true`.

File: modules/docker-build/versions.tf [hunk 2]
```diff
@@ -4,7 +4,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 3.35"
+      version = ">= 4.22.0"
```
Comment (antonbabenko) [inline modules/docker-build/versions.tf:7]:
```suggestion
      version = ">= 4.22"
```

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,9 @@
+# Environmental secrets
```
Comment (antonbabenko) [inline .gitignore:1]: Please revert these changes.

**Review by antonbabenko — CHANGES_REQUESTED** [review]

File: modules/docker-build/main.tf (hunk 1 above)
Comment (1Mill) [reply modules/docker-build/main.tf]: Added `var.ecr_force_delete` with a default value of `true`

File: .gitignore (hunk 3 above)
Comment (1Mill) [reply .gitignore]: Removed any changes to `.gitignore`

**Review by antonbabenko — APPROVED** [review]

**Comment by antonbabenko** [comment]
Hi @1Mill ! Thank you for this addition!

**Comment by antonbabenko** [comment]
This PR is included in version 4.0.0 :tada:
